rpc/kate-rpc/src/system.rs

```diff
@@ -217,6 +217,8 @@ where
 			}
 		}
 
+		cache.promote_block(block_hash);
```

Comment:
I think we can simply use the LRU cache instead of implementing our own custom cache